src/Http/Http/src/Internal/ResponseCookies.cs

```diff
@@ -88,6 +88,57 @@ public void Append(string key, string value, CookieOptions options)
             Headers[HeaderNames.SetCookie] = StringValues.Concat(Headers[HeaderNames.SetCookie], cookieValue);
         }
 
+        /// <inheritdoc />
+        public void Append(IDictionary<string, string> keyValuePairs, CookieOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            // SameSite=None cookies must be marked as Secure.
+            if (!options.Secure && options.SameSite == SameSiteMode.None)
+            {
+                if (_logger == null)
+                {
+                    var services = _features.Get<Features.IServiceProvidersFeature>()?.RequestServices;
```

Comment:
We want to log from these components like this.

Reply:
Thanks for the review, I kind of feel there is something missing after "like this", could you please tell me if I am missing something here?

Reply:
@davidfowl this was copied from the existing code above, we can discuss it separately from this PR.
https://github.com/dotnet/aspnetcore/blob/00b551ec3ca7b50cbfe7fc0cfc91fe9e51b37b99/src/Http/Http/src/Internal/ResponseCookies.cs#L59-L71

Reply:
Correct I did not fully look at this yet and just copied from the original append, I will definitely look if this can be improved too, thanks